cmake/macros/macro_check_mpi_interface.cmake

```diff
@@ -27,7 +27,7 @@ MACRO(CHECK_MPI_INTERFACE _feature _var)
     SET(_nope FALSE)
 
     FOREACH(_library ${${_feature}_LIBRARIES})
-      IF( _library MATCHES "/libmpi(|_cxx)\\.(a|so)[^/]*$")
+      IF( _library MATCHES "/libmpi\\.(a|so)[^/]*$")
```

Comment:
What about macOS dylib?

Reply:
@davydden Well, we haven't searched for that particular ending in the past. So I guess the check is useless on Macs. (I would prefer to completely revise the `CHECK_MPI_INTERFACE` macro in a separate PR.)

Reply:
Fine with me

Reply:
Fine with me

Reply:
If you want to generalize this, you can change the regex to use the `CMAKE_<TYPE>_LIBRARY_PREFIX` and  `CMAKE_<TYPE>_LIBRARY_SUFFIX` in the regex, with `<TYPE>` being `SHARED`, `STATIC` or `IMPORT` (e.g. on Windows). For example,  [CMAKE_SHARED_LIBRARY_SUFFIX](https://cmake.org/cmake/help/latest/variable/CMAKE_SHARED_LIBRARY_SUFFIX.html) will be `.so` and [CMAKE_SHARED_LIBRARY_PREFIX](https://cmake.org/cmake/help/latest/variable/CMAKE_SHARED_LIBRARY_PREFIX.html) will be `lib` on Linux. That should yield the most flexible solution.

Be aware though that not all MPIs are incompatible, e.g. there's the [MPICH ABI initiative](http://www.mpich.org/abi/).